pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -58,14 +57,11 @@ func (e *OpenTsdbExecutor) Execute(ctx context.Context, queries tsdb.QuerySlice,
 
 	tsdbQuery.Start = queryContext.TimeRange.GetFromAsMsEpoch()
 	tsdbQuery.End = queryContext.TimeRange.GetToAsMsEpoch()
+	tsdbQuery.Queries = make([]map[string]interface{}, len(queries))
 
-	for _, query := range queries {
-		tsdbQuery.Queries = []OpenTsdbMetric{
-			OpenTsdbMetric{
-				Metric:     query.Model.Get("metric").MustString(),
-				Aggregator: query.Model.Get("aggregator").MustString(),
-			},
-		}
+  for i := 0; i < len(queries); i++ {
+  	metric := e.buildMetric(queries[i])
+  	tsdbQuery.Queries[i] = metric
```

Comment:
Any reason for not appending metric to tsdbQuery.Queries? Ex `tsdbQuery.Queries = append(tsdbQuery.Queries, metric) `